File: DATN-Fashion-Shop-BE/src/main/java/com/example/DATN_Fashion_Shop_BE/dto/request/promotion/PromotionRequest.java
package com.example.DATN_Fashion_Shop_BE.dto.request.promotion;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PromotionRequest {
    private String description;
    private Double discountRate;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private List<Long> productIds;
}

File: FE/DATN-Fashion-Shop-FE/src/app/dto/promotionDTO/PromotionRequest.ts
export interface PromotionRequest {
  description: string;
  discountRate: number;
  startDate: string;  // ISO string (yyyy-MM-dd'T'HH:mm:ss)
  endDate: string;    // ISO string (yyyy-MM-dd'T'HH:mm:ss)
  productIds: number[];
}
